genesis/genesis.go

```diff
@@ -379,6 +376,177 @@ func (g *Genesis) activateDiscoveryNodes(ctx context.Context, cb *ContractsBuild
+	contract := core.NewRecordRef(*g.rootDomainRef.Record(), *nodeID)
+	_, err = g.ArtifactManager.ActivateObject(
+		ctx,
+		core.RecordRef{},
+		*contract,
+		*g.nodeDomainRef,
+		*cb.Prototypes[nodeRecord],
+		false,
+		nodeData,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ activateNodes ] Could'n activateNodeRecord node object")
+	}
+	_, err = g.ArtifactManager.RegisterResult(ctx, *g.rootDomainRef, *contract, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ activateNodes ] Couldn't register result to artifact manager")
+	}
+	return contract, nil
+}
+
+func (g *Genesis) addDiscoveryIndex(ctx context.Context, cb *ContractsBuilder, indexMap map[string]string) ([]genesisNode, map[string]string, error) {
+	errMsg := "[ addDiscoveryIndex ]"
+	discoveryKeysPath, err := abs(g.config.DiscoveryKeysDir)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, errMsg)
+	}
+	discoveryKeys, err := g.uploadKeys(ctx, discoveryKeysPath, len(g.config.DiscoveryNodes))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, errMsg)
+	}
+	discoveryNodes, err := g.activateDiscoveryNodes(ctx, cb, discoveryKeys)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, errMsg)
+	}
+	for _, node := range discoveryNodes {
+		indexMap[node.node.PublicKey] = node.ref.String()
+	}
+	return discoveryNodes, indexMap, nil
+}
+
+func (g *Genesis) addIndex(ctx context.Context, cb *ContractsBuilder, indexMap map[string]string) (map[string]string, error) {
+	errMsg := "[ addIndex ]"
+	nodeKeysPath, err := abs(g.config.NodeKeysDir)
+	if err != nil {
+		return nil, errors.Wrap(err, errMsg)
+	}
+	userKeys, err := g.uploadKeys(ctx, nodeKeysPath, nodeAmount)
+	if err != nil {
+		return nil, errors.Wrap(err, errMsg)
+	}
+	nodes, err := g.activateNodes(ctx, cb, userKeys)
+	if err != nil {
+		return nil, errors.Wrap(err, errMsg)
+	}
+	for _, node := range nodes {
+		indexMap[node.publicKey] = node.ref.String()
+	}
+	return indexMap, nil
+}
+
+func (g *Genesis) createKeys(ctx context.Context, path string, amount int) error {
+	err := os.RemoveAll(path)
+	if err != nil {
+		return errors.Wrap(err, "[ createKeys ] couldn't remove old dir")
+	}
+
+	for i := 0; i < amount; i++ {
+		ks := platformpolicy.NewKeyProcessor()
+
+		privKey, err := ks.GeneratePrivateKey()
+		if err != nil {
+			return errors.Wrap(err, "[ createKeys ] couldn't generate private key")
+		}
+
+		privKeyStr, err := ks.ExportPrivateKeyPEM(privKey)
+		if err != nil {
+			return errors.Wrap(err, "[ createKeys ] couldn't export private key")
+		}
+
+		pubKeyStr, err := ks.ExportPublicKeyPEM(ks.ExtractPublicKey(privKey))
+		if err != nil {
+			return errors.Wrap(err, "[ createKeys ] couldn't export public key")
+		}
+
+		result, err := json.MarshalIndent(map[string]interface{}{
+			"private_key": string(privKeyStr),
+			"public_key":  string(pubKeyStr),
+		}, "", "    ")
+		if err != nil {
+			return errors.Wrap(err, "[ createKeys ] couldn't marshal keys")
+		}
+
+		name := "node_" + strconv.Itoa(i) + ".json"
```

Comment:
Is this format used anywhere?
I mean "node-<id>.json".
I thing yes, that is why it would be better use format string for that